addons/stock/models/stock_quant.py

```diff
@@ -1347,12 +1347,12 @@ def _check_serial_number(self, product_id, lot_id, company_id, source_location_i
             if quants:
                 if not source_location_id:
                     # trying to assign an already existing SN
-                    message =  _('The Serial Number (%s) is already used in these location(s): %s.\n\n'
-                                 'Is this expected? For example this can occur if a delivery operation is validated '
-                                 'before its corresponding receipt operation is validated. In this case the issue will be solved '
-                                 'automatically once all steps are completed. Otherwise, the serial number should be corrected to '
-                                 'prevent inconsistent data.',
-                                 lot_id.name, ', '.join(sn_locations.mapped('display_name')))
+                    message = _('The Serial Number (%(lot)s) is already used in these location(s): %(location_list)s.\n\n'
+                                'Is this expected? For example this can occur if a delivery operation is validated '
+                                'before its corresponding receipt operation is validated. In this case the issue will be solved '
+                                'automatically once all steps are completed. Otherwise, the serial number should be corrected to '
+                                'prevent inconsistent data.',
+                                lot=lot_id.name, location_list=format_list(self.env, sn_locations.mapped('display_name')))
```

Comment:
jeez, who originally added this :hankey:  (hint, it was me :speak_no_evil: )

```suggestion
                    message = _('The Serial Number (%(serial_number)s) is already used in location(s): %(location_list)s.\n\n'
                                'Is this expected? For example, this can occur if a delivery operation is validated '
                                'before its corresponding receipt operation is validated. In this case the issue will be solved '
                                'automatically once all steps are completed. Otherwise, the serial number should be corrected to '
                                'prevent inconsistent data.',
                                serial_number=lot_id.name, location_list=format_list(self.env, sn_locations.mapped('display_name')))
```